src/lib/modules/blockchain_connector/provider.js

```diff
@@ -16,9 +20,48 @@ class Provider {
 
   startWeb3Provider(callback) {
     const self = this;
-
+    
     if (this.type === 'rpc') {
       self.provider = new this.web3.providers.HttpProvider(self.web3Endpoint);
+    } else if (this.type === 'ledger') {
+        let hardwareWalletConfigs = [];
+
+        self.accountsConfig.forEach((accountConfig) => {
+          if (accountConfig.hardwareWallet) {
+            if (accountConfig.hardwareWallet === 'ledger') {
+              hardwareWalletConfigs.push(accountConfig.hardwareWallet);
+            }
+          }
+        });
+
+        if (hardwareWalletConfigs.length > 0) {
```

Comment:
To make it clearer and cleaner, you could do:
```
if (!hardwareWalletConfigs.length) {
  // TODO when we have documentation for it, add a link in the error message
  return callback(__('No ledger wallet config found, please add it'));
}
```
Notice how I return and callback immediately. That's because, even if we went through with the initialization, it would bug because thee is no account.
Also, that way, you don't need an `else`